sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -1022,7 +1022,16 @@ case class CacheTableAsSelect(
     plan: LogicalPlan,
     originalText: String,
     isLazy: Boolean,
-    options: Map[String, String]) extends LeafCommand
+    options: Map[String, String],
+    isAnalyzed: Boolean = false) extends AnalysisOnlyCommand {
+  override protected def withNewChildrenInternal(
+      newChildren: IndexedSeq[LogicalPlan]): CacheTableAsSelect =
+    copy(plan = newChildren.head)
```

Comment:
is it possible that `newChildren` is empty? Probably safer to add `if (isAnalyzed) ... else ...`

Reply:
This should be called only when there exist children. Maybe an assert is better `assert(!isAnalyzed)`? WDYT?

https://github.com/apache/spark/blob/e40fce919ab77f5faeb0bbd34dc86c56c04adbaa/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala#L345-L350

Reply:
assert SGTM